node/repo/fsrepo.go

```diff
@@ -486,7 +487,7 @@ func (fsr *fsLockedRepo) Put(name string, info types.KeyInfo) error {
 	keyPath := fsr.join(fsKeystore, encName)
 
 	_, err := os.Stat(keyPath)
-	if err == nil {
+	if err == nil && !strings.HasPrefix(name, wallet.KTrashPrefix) {
```

Comment:
This will go on to overwrite the existing trash key file in the case described in #4621, when we could just return at this point. I'd suggest doing: 

```suggestion
	if err == nil && strings.HasPrefix(name, wallet.KTrashPrefix) {
	    // Fine to try to write the same trash-prefixed file multiple times
	    return nil
	} else if err == nil {
	    return xerrors.Errorf("checking key before put '%s': %w", name, types.ErrKeyExists)
	} else if !os.IsNotExist(err) {
		return xerrors.Errorf("checking key before put '%s': %w", name, err)			
	}
```